polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -685,22 +687,38 @@ async fn advertise_collation<Context>(
 				.validator_group
 				.should_advertise_to(candidate_hash, peer_ids, &peer);
 
-		if !should_advertise {
-			gum::debug!(
-				target: LOG_TARGET,
-				?relay_parent,
-				peer_id = %peer,
-				"Not advertising collation since validator is not interested",
-			);
-			continue
+		match should_advertise {
+			ShouldAdvertiseTo::Yes => {},
+			ShouldAdvertiseTo::NotAuthority => {
+				gum::trace!(
+					target: LOG_TARGET,
+					?relay_parent,
+					?candidate_hash,
+					peer_id = %peer,
+					"Not advertising collation: not relevant to peer"
+				);
+				continue
+			},
+			ShouldAdvertiseTo::AlreadyAdvertised => {
+				gum::debug!(
+					target: LOG_TARGET,
+					?relay_parent,
+					?candidate_hash,
+					peer_id = %peer,
+					"Not advertising collation: already advertised"
+				);
+				continue
+			},
```

Comment:
if we do `#[derive(Debug, PartialEq)]` on `ShouldAdvertiseTo`, we can make this shorter:
```suggestion
        if should_advertise != ShouldAdvertiseTo::Yes {
                gum::trace!(
					target: LOG_TARGET,
					?relay_parent,
					?candidate_hash,
					peer_id = %peer,
					reason = ?should_advertise,
					"Not advertising collation
				);
				continue
        }
```

Reply:
Would be the same as using a wildcard `_` .. no?

Reply:
Logs are different based on the outcome

Reply:
Yeah, the point was to use `Debug` impl to differentiate the different logs.

Reply:
but yeah, now they're both either debug/trace.

Reply:
@sandreim add the `reason` to the logs to take care of that.